src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -24,49 +44,85 @@ class F extends ValidationTest {
     });
   }
 
-  createRenderPipeline(): GPURenderPipeline {
+  createRenderPipelineWithLayout(
+    device: GPUDevice,
+    bindGroups: Array<Array<GPUBindGroupLayoutEntry>>
+  ): GPURenderPipeline {
+    const shader = `
+      [[stage(vertex)]] fn vs_main() -> [[builtin(position)]] vec4<f32> {
+        return vec4<f32>(1.0, 1.0, 0.0, 1.0);
+      }
+
+      [[stage(fragment)]] fn fs_main() -> [[location(0)]] vec4<f32> {
+        return vec4<f32>(0.0, 1.0, 0.0, 1.0);
+      }
+    `;
+    const bindGroupLayouts = getBindGroupLayouts(this.device, bindGroups);
+
     const pipeline = this.device.createRenderPipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts }),
       vertex: {
         module: this.device.createShaderModule({
-          code: `
-            [[block]] struct VertexUniforms {
-              transform : mat2x2<f32> ;
-            };
-            [[group(0), binding(0)]] var<uniform> uniforms : VertexUniforms;
-
-            [[stage(vertex)]] fn main(
-              [[builtin(vertex_index)]] VertexIndex : u32
-              ) -> [[builtin(position)]] vec4<f32> {
-              var pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
-                vec2<f32>(-1.0, -1.0),
-                vec2<f32>( 1.0, -1.0),
-                vec2<f32>(-1.0,  1.0)
-              );
-              return vec4<f32>(uniforms.transform * pos[VertexIndex], 0.0, 1.0);
-            }`,
+          code: shader,
         }),
-        entryPoint: 'main',
+        entryPoint: 'vs_main',
       },
       fragment: {
         module: this.device.createShaderModule({
```

Comment:
createShaderModule only needs to be called once. (both calls have the same args)